crypto/types/compact_bit_array.go

```diff
@@ -75,13 +75,22 @@ func (bA *CompactBitArray) SetIndex(i int, v bool) bool {
 // there are two bits set to true before index 4.
 func (bA *CompactBitArray) NumTrueBitsBefore(index int) int {
 	numTrueValues := 0
-	for i := 0; i < index; i++ {
-		if bA.GetIndex(i) {
-			numTrueValues++
+	max := bA.Count()
+	if index > max {
```

Comment:
Shouldn't we panic in this case, its invalid API usage?

Reply:
I think we should

Reply:
I don't think it's invalid. I don't want to access any value of of the array. Let's say array length is 100, asking for true bits before 100 is same as asking about 200. Asking for longer range shouldn't break the function IMHO.
Moreover the previous implementation also didn't panic.

Reply:
Hold on - what was the behaviour before this change?

Reply:
Same, but less optimal -> see tests.